lib/core/src/Cardano/Wallet/Gen.hs

```diff
@@ -225,19 +225,32 @@ shrinkByteString bs
   where
     n = BS.length bs
 
-genText :: Gen Text
-genText =
-    (T.pack . take 32 . getPrintableString <$> arbitrary) `suchThat` guardText
+genTxMetaText :: Gen Text
+genTxMetaText =
+    (T.pack . take 32 . getPrintableString <$> arbitrary)
+    `suchThat` guardTxMetaText
 
-shrinkText :: Text -> [Text]
-shrinkText t
+shrinkTxMetaText :: Text -> [Text]
+shrinkTxMetaText t
     | n <= 1    = []
-    | otherwise = filter guardText [ T.take (n `div` 2) t, T.drop (n `div` 2) t ]
+    | otherwise = filter guardTxMetaTextPrefix
+        [ T.take (n `div` 2) t, T.drop (n `div` 2) t ]
```

Comment:
I'm pretty sure `guardText` isn't needed anymore - a relic of a previous JSON encoding of TxMetadata. So we could just re-use the `UnicodeString` shrinker.

```haskell
shrinkTxMetaText t = [T.pack s | u <- shrink (UnicodeString $ T.unpack t), let s = getUnicodeString u, length s > 0]
```

Reply:
Fixed in ab6f53b4639fa56cf6c4bc714948d7d0b31e2ebe and 58cc2af86cf7d3029d4676963acf065d7b63e18b.